pkg/reconciler/trustroot/trustroot.go

```diff
@@ -191,3 +181,81 @@ func (r *Reconciler) removeTrustRootEntry(ctx context.Context, cm *corev1.Config
 	}
 	return nil
 }
+
+// pemToKeyAndID takes a public key in PEM format, and turns it into
+// crypto.PublicKey and the CTLog LogID.
+func pemToKeyAndID(pem []byte) (crypto.PublicKey, string, error) {
+	pk, err := cryptoutils.UnmarshalPEMToPublicKey(pem)
+	if err != nil {
+		return nil, "", fmt.Errorf("unmarshaling PEM public key: %w", err)
+	}
+	logID, err := ctutil.GetCTLogID(pk)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to construct LogID for rekor: %w", err)
+	}
+	return pk, hex.EncodeToString(logID[:]), nil
+}
+
+// These are private to sigstore/sigstore even though I don't think they should
+// be.
+type customMetadata struct {
+	Usage  sigstoretuf.UsageKind  `json:"usage"`
+	Status sigstoretuf.StatusKind `json:"status"`
+}
+
+type sigstoreCustomMetadata struct {
+	Sigstore customMetadata `json:"sigstore"`
+}
+
+// getSigstoreKeysFromTuf returns the sigstore keys from the TUF client. Note
+// that this should really be exposed from the sigstore/sigstore TUF pkg, but
+// is currently not.
+func getSigstoreKeysFromTuf(ctx context.Context, tufClient *client.Client) (*config.SigstoreKeys, error) {
+	targets, err := tufClient.Targets()
+	if err != nil {
+		return nil, fmt.Errorf("error getting targets: %w", err)
+	}
+	ret := &config.SigstoreKeys{}
+	for name, targetMeta := range targets {
+		// Skip any targets that do not include custom metadata.
+		if targetMeta.Custom == nil {
+			continue
+		}
+		var scm sigstoreCustomMetadata
+		err := json.Unmarshal(*targetMeta.Custom, &scm)
+		if err != nil {
+			logging.FromContext(ctx).Errorf("**Warning** Custom metadata not configured properly for target %s, skipping target: %v", name, err)
```

Comment:
This is logged as an error but it looks more of a Warning. What about using Warnf here ?